api4/schema.graphqls

```diff
@@ -0,0 +1,157 @@
+schema {
+  query: Query
+}
+
+type Query {
+	user(id: String!): User
+	config(): StringMap!
+	license(): StringMap!
+	teamMembers(userId: String!,
+		teamId: String = ""): [TeamMember]!
+	channels(userId: String!,
+		teamId: String = "",
+		includeDeleted: Boolean = false,
+		lastDeleteAt: Float = 0,
+		lastUpdateAt: Float = 0,
+		first: Int = 60,
+		after: String = ""): [Channel]!
+	channelsLeft(userId: String!,
+		since: Float!): [String!]!
+	channelMembers(userId: String!,
+		channelId: String = "",
+		first: Int = 60,
+		after: String = "",
+		lastUpdateAt: Float = 0): [ChannelMember]!
+}
+
+scalar ChannelType
+
+scalar SidebarCategoryType
+
+scalar SidebarCategorySorting
+
+scalar StringMap
+
+scalar Time
+
+type Channel {
+	id           : String!
```

Comment:
Yeah, good point. Actually these ids are basically the ids we have in the DB for an object. But the `ID` scalar type is meant as an opaque value not intended to be human-readable, and moreover, it's meant to serve as a globally unique identifier for the object and satisfy the node interface.

```
# An object with a Globally Unique ID
interface Node {
  # The ID of the object.
  id: ID!
}
```

So essentially, this is going to be the cursor field for an object, because the cursor is a base64 encoded value containing the object name and it's ID, so that on the server, we can decode the value, to get the object type, and then query the appropriate table for that value.

So this `id` field now will need to change to `objectId` or something and the cursor field will be replaced by the `id`. I'll do it after this PR.

Reply:
Our IDs are globally unique (hopefully) so we could use our `id` as the graphql `ID` except that we need the cursor to be scoped to the object type? Why does the cursor need to be scoped? Doesn't context always tell you what object type you are dealing with?

Reply:
> Our IDs are globally unique (hopefully)

There's no guarantee of that :stuck_out_tongue: 

> Why does the cursor need to be scoped? Doesn't context always tell you what object type you are dealing with?

So if an id is global, what a client can essentially do is ask a server to give an object for an id. In that case, the server won't know which table to pick from. That's why we need to add the object name to the id.

Reply:
Ah so graphql clients can ask for objects just by global id. That makes sense. 
Thanks!